### ISSUE TITLE
Reference includes ADDRESS/LOCATION for @BOOK/@INBOOK if provided

### ISSUE THREAD

**Issue by nilsreimer** [plk/biblatex-apa#95]
Apologies if I misunderstand how the package is supposed to work. Suppose I want to cite Example 20 from the manual and want to include an address in the .bib file for backward compatibility:

```
@BOOK{10.2:20,
  AUTHOR         = {L. S. Brown},
  TITLE          = {Feminist Therapy},
  EDITION        = {2},
  ADDRESS        = {Washington, DC},
  PUBLISHER      = {American Psychological Association},
  DATE           = {2018},
  DOI            = {10.1037/0000092-000}
}
```
I cite this reference as usual:
```
\documentclass{article}
\usepackage[american]{babel}
\usepackage{csquotes}
\usepackage[style=apa]{biblatex} 
\addbibresource{example.bib}
\begin{document}
\autocite{10.2:20}.
\printbibliography
\end{document}
```
I get a reference that (incorrectly) includes the location of the publisher:
> Brown, L. S. (2018). _Feminist therapy_ (2nd ed.). Washington, DC, American Psychological Association. https://doi.org/10.1037/0000092-000

Can I get biblatex to not include publisher locations in the reference list, even if they are provided in the .bib file, or do I have to manually remove publisher locations from the .bib file? Thank you in advance.

**Comment by plk** [comment]
This will be fixed in v9.8 which will be released in the next couple of days. It only really concerns `book` entries as other types of entries in APA 7 do include the location information. If you want to get rid of it for books until v9.8 is released, put this in your preamble:

```
\DeclareSourcemap{
  \maps[datatype=bibtex]{
    \map{
      \pertype{book}
      \step[fieldset=location, null]
    }
  }
}
```

**Comment by nilsreimer** [comment]
Thank your for your swift response—and for maintaining this package!

**Comment by moewew** [comment]
I'm wondering about the other entry types. (I don't have access to the manual.) It would seem odd to suppress the `location` for `@book` and `@collection` but not for `@inbook`/`@incollection`.
```latex
\documentclass[british]{article}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{babel}
\usepackage{csquotes}

\usepackage[style=apa, backend=biber]{biblatex}

\addbibresource{biblatex-examples.bib}


\begin{document}
\autocite{pines,kant:ku}
\printbibliography
\end{document}
```

The *Quick Reference Guide* available at https://apastyle.apa.org/instructional-aids/reference-guide.pdf asks not to include the `location` for all `@book`/`@collection` as well as `@inbook`/`@incollection`.

**Comment by nilsreimer** [comment]
It might also be worthwhile to suppress `address` which is used in the output of some reference managers instead of location.

**Comment by plk** [comment]
biber maps `address` to `location` internally anyway ...

**Comment by moewew** [comment]
@nilsreimer `address` shouldn't be an issue as that field is automatically mapped to `location`. So if the `location` is not printed, neither is `address`. You only need to take care of `address` seprately, if you use a sourcemap (e.g. the one shown by PLK above), but since `apa.bbx` uses another method to suppress the `location` (namely not to print it), that isn't relevant.

**Comment by plk** [comment]
I missed `inbook` - I will change that too.

**Comment by moewew** [comment]
OK. I couldn't find anything official on other types,  but the second-hand accounts that I read all strongly suggest that `location` is out for pretty much all entry types. (I guess it is hard to prove a negative by examples, so ...)

**Comment by plk** [comment]
It is used for some `misc`, `unpublished` and `proceedings` types, plus special treatment in the rather non-standard legal types but it is absent in the main types, true.

**Comment by moewew** [comment]
I noticed it us uses in `periodical`, which looks a bit odd to me, since `@periodical` is the type for a journal and `@article`s (if you will "`@inperiodical`s") never show the `location` (or a publisher).

For `@proceedings` (`@inproceedings`) I'm wondering if `venue` might not be more appropriate. The standard styles group `eventtitle`, `eventdate` and `venue`. `location` is always the location of the `publisher` (or equivalent `institution`, `organization`).

**Comment by plk** [comment]
The `location` fields in APA `proceedings` and `inproceedings` look a lot like the standard `location` fields for publishers ... they also contain the cite/state/country in the same format as for publishers. `venue` to me is more specific than this, like "So and so conference centre"?

**Comment by moewew** [comment]
Mhhh, I agree that `venue` can definitely be more specific than just a city. The only example of `venue` in `biblatex-examples.bib` is
```latex
@inproceedings{salam,
  author         = {Salam, Abdus},
  editor         = {Svartholm, Nils},
  title          = {Weak and Electromagnetic Interactions},
  date           = 1968,
  booktitle      = {Elementary particle theory},
  booksubtitle   = {Relativistic groups and analyticity},
  booktitleaddon = {Proceedings of the {Eighth Nobel Symposium}},
  eventdate      = {1968-05-19/1968-05-25},
  venue          = {Aspen{\"a}sgarden, Lerum},
  publisher      = {Almquist \& Wiksell},
  location       = {Stockholm},
  pages          = {367-377},
}
```
where I think "Lerum" is a village and Aspenäs might have been a hotel?

What's important for `@(in)procceedings` is that `location` *always* refers to the `publisher`, so if you want to give the conference location, `venue` is the only viable option. This is in contrast to standard BibTeX styles, where `address` is the location of the publisher except for `@inproceedings` where it is the location of the conference. `biblatex` is more consistent there: `location` is always the `publisher` (or equivalent) location.

The `biblatex` docs say
> [`venue`] The location of a conference, a symposium, or some other event in `@proceedings`
and `@inproceedings` entries. This field may also be useful for the custom types
listed in § 2.1.3. Note that the `location` list holds the place of publication. It
therefore corresponds to the `publisher` and `institution` lists. The `location`
of the event is given in the `venue` field. See also `eventdate` and `eventtitle`.

I did some more research and all second-hand accounts of APA 7th edition that I found suggest that there is *no* location information for conference papers published in book-type `@proceedings` or in a journal. (e.g. https://libguides.jcu.edu.au/apa/articles/conference-papers, http://libguides.lub.lu.se/c.php?g=676676&p=4821142).

APA style appears to want a conference location for unpublished (or formally unpublished) conference material such as sessions or talks: http://writeanswers.royalroads.ca/faq/199260. I would probably want to argue that this material constitutes an entry type separate from `@(in)proceedings`. I would still use `eventtitle`+`venue` there, but I could probably be persuaded to accept `location` here.
